### PR TITLE
NRF52: add ADC and PWM support

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary
- drivers/timers/pwm.c: add missing new lines to some debug messages 
- nrf52: add PWM support
- nrf52: add ADC support
- boards/nrf52840-dk: add ADC and PWM examples

## Testing
nrf52840-dk

### CODE REVIEW

File: arch/arm/src/nrf52/nrf52_pwm.c [hunk 1]
```diff
@@ -0,0 +1,719 @@
...
+        {
+          lower = &g_nrf52_pwm;
+          break;
+        }
+#endif
+
+#ifdef CONFIG_NRF52_PWM3
+      case 3:
+        {
+          lower = &g_nrf52_pwm;
```
Comment (acassis) [inline arch/arm/src/nrf52/nrf52_pwm.c:704]: g_nrf52_pwm3

File: arch/arm/src/nrf52/nrf52_pwm.c [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+        {
+          lower = &g_nrf52_pwm1;
+          break;
+        }
+#endif
+
+#ifdef CONFIG_NRF52_PWM2
+      case 2:
+        {
+          lower = &g_nrf52_pwm;
```
Comment (acassis) [inline arch/arm/src/nrf52/nrf52_pwm.c:696]: g_nrf52_pwm2

Comment (raiden00pl) [reply arch/arm/src/nrf52/nrf52_pwm.c]: done

File: arch/arm/src/nrf52/nrf52_pwm.c (hunk 1 above)
Comment (raiden00pl) [reply arch/arm/src/nrf52/nrf52_pwm.c]: done

**Comment by acassis** [comment]
LGTM, thank you!

**Pull request merged**